procdiscovery/pkg/inspectors/nginx/nginx.go

```diff
@@ -13,12 +15,40 @@ type NginxInspector struct{}
 
 const NginxProcessName = "nginx"
 
-func (j *NginxInspector) Inspect(p *process.Details) (common.ProgramLanguageDetails, bool) {
-	var programLanguageDetails common.ProgramLanguageDetails
+func (j *NginxInspector) Inspect(p *process.Details) (common.ProgrammingLanguage, bool) {
 	if strings.Contains(p.CmdLine, NginxProcessName) || strings.Contains(p.ExeName, NginxProcessName) {
-		programLanguageDetails.Language = common.NginxProgrammingLanguage
-		return programLanguageDetails, true
+		return common.NginxProgrammingLanguage, true
 	}
 
-	return programLanguageDetails, false
+	return "", false
+}
+
+func (j *NginxInspector) GetRuntimeVersion(p *process.Details, podIp string) string {
+	version, err := GetNginxVersion(podIp)
+	if err != nil {
+		return ""
+	}
+
+	return version
+}
+
+func GetNginxVersion(podIP string) (string, error) {
+	resp, err := http.Get("http://" + podIP)
+	if err != nil {
+		return "", nil
+	}
+	defer resp.Body.Close()
+
+	serverHeader := resp.Header.Get("Server")
```

Comment:
This may be a potential security issue (a malicious server returns a 1000Gb header for example) can we read only the first X chars?

Reply:
I'm reading a specific key in this call. You think that the Value of this key in header can be big? Or all the Header?